### ISSUE TITLE
Register-ArgumentCompleter doesnt give completions in the editor

### ISSUE THREAD

**Issue by PrzemyslawKlys** [PowerShell/vscode-powershell#2162]
Issue Description
=====

Register-AutoCompleter doesn't seem to work properly in VSCode. It only works for single string. 

```powershell
function Test-Meteor {
    param(
        [validateset('Test','Test2','Test3')][string[]] $Value,
        [string[]] $OtherValue
    )


}


[scriptblock] $Test = {
    param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameters)

    @('Test','Test1','Test2')
}

Register-ArgumentCompleter -CommandName Test-Meteor -ParameterName OtherValue -ScriptBlock $Test


Test-Meteor -Value Test,Test2 -OtherValue Test1
```

![image](https://user-images.githubusercontent.com/15063294/64082110-68047380-cd0a-11e9-98c8-481f27f2aa54.png)

This works fine when you try in VScode Terminal (even the Powershell Integrated one) or outside of VSCode. Sometimes it starts working but only for 1st value, 2nd value is loaded from cache along with some garbage. 

Attached Logs
=====

Follow the instructions in the [README](https://github.com/PowerShell/vscode-powershell%23reporting-problems) about
capturing and sending logs.

Environment Information
=====

Visual Studio Code
-----

| Name | Version |
| --- | --- |
| Operating System | Windows_NT x64 10.0.18362 |
| VSCode | 1.37.1|
| PowerShell Extension Version | 2019.5.0 |

PowerShell Information
-----

|Name|Value|
|---|---|
|PSVersion|5.1.18362.145|
|PSEdition|Desktop|
|PSCompatibleVersions|1.0 2.0 3.0 4.0 5.0 5.1.18362.145|
|BuildVersion|10.0.18362.145|
|CLRVersion|4.0.30319.42000|
|WSManStackVersion|3.0|
|PSRemotingProtocolVersion|2.3|
|SerializationVersion|1.1.0.1|

Visual Studio Code Extensions
-----

<details><summary>Visual Studio Code Extensions(Click to Expand)</summary>

|Extension|Author|Version|
|---|---|---|
|auto-rename-tag|formulahendry|0.1.0|
|beautify|HookyQR|1.5.0|
|bracket-pair-colorizer-2|CoenraadS|0.0.28|
|code-settings-sync|Shan|3.4.2|
|githistory|donjayamanne|0.4.6|
|gitlens|eamodio|9.9.3|
|html-css-class-completion|Zignd|1.19.0|
|live-html-previewer|hdg|0.3.0|
|markdown-all-in-one|yzhang|2.4.2|
|markdown-preview-enhanced|shd101wyy|0.4.3|
|markdown-shortcuts|mdickin|0.11.0|
|material-icon-theme|PKief|3.9.0|
|powershell|ms-vscode|2019.5.0|
|project-manager|alefragnani|10.7.0|
|theme-monokai-pro-vscode|monokai|1.1.14|
|vscode-html-css|ecmel|0.2.3|
|vscode-wakatime|WakaTime|2.2.0|
|xml|DotJoshJohnson|2.5.0|

**Comment by SydneyhSmith** [comment]
@PrzemyslawKlys thanks for opening this bug and providing detailed repro steps--we were able to reproduce this issue...we will have to do some more investigation as to where the connection is missing to provide the editor with this context

**Comment by PrzemyslawKlys** [comment]
Just so you know the same problem most likely relates to:

```
        [ArgumentCompleter(
            {
                param($CommandName, $ParameterName, $WordToComplete, $CommandAst, $FakeBoundParameters)
                ((Get-EventsDefinitions).Keys)
            }
        )]
        [ValidateScript(
            {
                $_ -in (((Get-EventsDefinitions).Keys))
            }
        )]
```
I remember having issues with it. It also impacts dynamic parameters.

```
    DynamicParam {
        # Defines Report / Dates Range dynamically from HashTables and saved files
        $ParameterSetsAttributesDateManual = New-Object System.Management.Automation.ParameterAttribute
        $ParameterSetsAttributesDateManual.Mandatory = $true
        $ParameterSetsAttributesDateManual.ParameterSetName = 'DateManual'

        $ParamAttribDatesRange = New-Object System.Management.Automation.ParameterAttribute
        $ParamAttribDatesRange.Mandatory = $true
        $ParamAttribDatesRange.ParameterSetName = 'DateRange'

        $ParameterSetsAttributes = New-Object System.Management.Automation.ParameterAttribute
        $ParameterSetsAttributes.Mandatory = $true
        $ParameterSetsAttributes.ParameterSetName = 'Manual'

        # Definitions for Report
        $Names = (Get-EventsDefinitions).Keys
        $ReportAttrib = New-Object  System.Collections.ObjectModel.Collection[System.Attribute]
        $ReportAttrib.Add($ParameterSetsAttributes)
        $ReportAttrib.Add($ParamAttribDatesRange)
        $ReportAttrib.Add($ParameterSetsAttributesDateManual)
        $ReportAttrib.Add((New-Object System.Management.Automation.ValidateSetAttribute($Names)))
        $ReportRuntimeParam = New-Object System.Management.Automation.RuntimeDefinedParameter('Report', [string[]], $ReportAttrib)

        # Definitions for Dates Range
        $DatesRange = (Get-DatesDefinitions -Skip 'CustomDate', 'CurrentDayMinuxDaysX', 'CurrentDayMinusDayX', 'OnDay')
        $DatesRangeAttrib = New-Object  System.Collections.ObjectModel.Collection[System.Attribute]
        $DatesRangeAttrib.Add($ParamAttribDatesRange)
        $DatesRangeAttrib.Add((New-Object System.Management.Automation.ValidateSetAttribute($DatesRange)))
        $DatesRangeRuntimeParam = New-Object System.Management.Automation.RuntimeDefinedParameter('DatesRange', [string], $DatesRangeAttrib)

        # Finalization
        $RuntimeParamDic = New-Object System.Management.Automation.RuntimeDefinedParameterDictionary
        $RuntimeParamDic.Add('Report', $ReportRuntimeParam)
        $RuntimeParamDic.Add('DatesRange', $DatesRangeRuntimeParam)
        return $RuntimeParamDic
    }
```

Sorry that I didn't provide a "test" but if it's needed I can provide some minified versions.

**Comment by M1kep** [comment]
Could this introduce a security concern of code running simply by attempting to do ArgumentCompletion within VSCode editor? Should the editor environment be considered a "no run" area, unless explicitly requested?

**Comment by M1kep** [comment]
Related? #831

**Comment by M1kep** [comment]
When the code is run in the Integrated Console it looks like the Intellisense engine "overwrites" or "prefers" the autocompletion with the in-file definition. Which means that the values aren't known due to the issue stated in #831

**Comment by SydneyhSmith** [comment]
Closing as fixed by PowerShell/PowerShell#10574
